tests/unit_tests/async_events/async_query_manager_tests.py

```diff
@@ -26,17 +27,27 @@
     AsyncQueryManager,
     AsyncQueryTokenException,
 )
+from superset.async_events.cache_backend import (
+    RedisCacheBackend,
+    RedisSentinelCacheBackend,
+)
 
 JWT_TOKEN_SECRET = "some_secret"
 JWT_TOKEN_COOKIE_NAME = "superset_async_jwt"
 
+# Define the cache backends once as mocks
+cache_backends = {
+    "RedisCacheBackend": mock.Mock(spec=RedisCacheBackend),
+    "RedisSentinelCacheBackend": mock.Mock(spec=RedisSentinelCacheBackend),
+    "redis.Redis": mock.Mock(spec=redis.Redis),
+}
+
```

Comment:
Could this loop be replaced by `pytest.mark.parametrize`? It feels more idiomatic (unless I'm missing some reason why it needs to be implemented like this..)

Reply:
Yes, that's a great suggestion. I'll update the code with `pytest.mark.parametrize` and commit the changes.

Reply:
Using `pytest.mark.parametrize` with `mock` from `unittest` caused compatibility issues. Switching to `parameterized.expand` resolved these issues and successfully covered the test cases, leading to a successful CI run. I’ll revisit this case later for further investigation.

Reply:
@nsivarajan I believe it's generally suggested to use `pytest_mock` with `pytest` test cases. I don't want to drag out this review process more than necessary, but I think the cleanest solution would be one with `pytest.mark.parametrize` and `pytest_mock`. LMKWYT, but in the meantime let me re-review the functional parts.

Reply:
Thanks, @villebro , for the review. I agree that using the standard approach is ideal. However, I've encountered continuous failures when combining `pytest.mark.parametrize` with `mock`, which suggests there might be compatibility issues. After some research, I found references that indicate potential conflicts when using pytest features with unittest.TestCase subclasses and mock (see [pytest documentation](https://docs.pytest.org/en/stable/how-to/unittest.html#pytest-features-in-unittest-testcase-subclasses), [SeleniumBase issue](https://github.com/seleniumbase/SeleniumBase/issues/395), and [pytest issue](https://github.com/pytest-dev/pytest/issues/541#issuecomment-609956951)).

I noticed that our repository already uses `parameterized.expand` in similar cases, so I opted for it here to ensure the tests run smoothly.

Would it be possible to proceed with this PR using `parameterized.expand`? I'll continue investigating the issues with `pytest.mark.parametrize` and `mock` and plan to submit a follow-up PR if I find a viable solution.

Reply:
Thanks for putting in the extra effort on these tests @nsivarajan ! Let's not let this derail your PR - I can also take a stab at migrating the remaining components to `pytest` (seems like a fun coffee break challenge).